### PR TITLE
One ruleset

### PR DESCRIPTION
**Pull request by nulltoken**
Contributes to #561

**Checklist**

- [ ] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

Fueled by https://github.com/stoplightio/spectral/pull/677#issuecomment-541444985 and the fact that oas2 and aos3 rulesets are doomed, this PR aims at attempting to: 
- [ ] reducing the build pipeline complexity
- [ ] make explicit that `oas2` and `oas3` are only temporary aliases to `oas`
- [ ] align the /src structure to this goal

### CODE REVIEW

File: src/rulesets/map.ts [hunk 1]
```diff
@@ -4,6 +4,6 @@ function resolveSpectralRuleset(ruleset: string) {
 
 export const filesMap = new Map<string, string>([
   ['spectral:oas', resolveSpectralRuleset('oas')],
-  ['spectral:oas2', resolveSpectralRuleset('oas2')],
-  ['spectral:oas3', resolveSpectralRuleset('oas3')],
+  ['spectral:oas2', resolveSpectralRuleset('oas')],
+  ['spectral:oas3', resolveSpectralRuleset('oas')],
```
Comment (P0lip) [inline src/rulesets/map.ts]: It's a breaking release, so you can remove both of them.
```suggestion
```

File: src/rulesets/__tests__/finder.karma.test.ts [hunk 2]
```diff
@@ -10,13 +10,13 @@ describe('Rulesets finder', () => {
   for (const shorthand of ['oas', 'oas2', 'oas3']) {
```
Comment (P0lip) [inline src/rulesets/__tests__/finder.karma.test.ts]:
```suggestion
```
Not going to be needed. A single test case will be sufficient

**Comment by philsturgeon** [comment]
How is this one coming along? I'm excited to see oas2 and oas3 deleted :)

**Comment by nulltoken** [comment]
> How is this one coming along? I'm excited to see oas2 and oas3 deleted :)

Damn! Time flies like an arrow.. Sorry @philsturgeon, I've been swamped in other $DAYJOB related topics that dragged me away from OSS.

I'll try and revive this in the following days.

Thanks for the kind bump.

**Comment by philsturgeon** [comment]
@nulltoken Another 500 trees if this can be done in the next week :D

**Comment by nulltoken** [comment]
> @nulltoken Another 500 trees if this can be done in the next week :D

@philsturgeon Playing with my carbon footprint guilt, aren't you? :stuck_out_tongue_winking_eye: 

Just a quickie, just saw you started working on #772 which will likely result in conflicts and/or duplicated work. How would you like us to proceed?

**Comment by nulltoken** [comment]
> > @nulltoken Another 500 trees if this can be done in the next week :D
> 
> @philsturgeon Playing with my carbon footprint guilt, aren't you? stuck_out_tongue_winking_eye
> 
> Just a quickie, just saw you started working on #772 which will likely result in conflicts and/or duplicated work. How would you like us to proceed?

Closed in favor of #772.

Opened #773 to help nail #772 faster.

**Pull request closed without merging**